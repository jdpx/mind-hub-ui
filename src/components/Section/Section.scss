@import "../../styles/colours.scss";

.content-wrapper {
    display: flex;
    flex: 1;
    border: 5px solid $background-blue;
    box-shadow: 3px 3px 4px rgba(36, 59, 113, 0.42);
    border-radius: 20px;
    margin: 15px;
    flex-direction: row;

    &.sm {
        max-width: 284px;
    }
    &.md {
        max-width: 466px;
    }
    &.lg {
        max-width: 1216px;
    }
    &.xl {
        width: 100%;
    }

    &.disabled {
        color: $disabled-text-grey;
        border-color: $disabled-border-grey;
        box-shadow: 3px 3px 10px #e3d7e7;
    }

    &.withTitle {
        flex-direction: column;
    }
}

.section-title {
    margin-left: 10px;
    margin-bottom: 0px;
    padding: 15px;
}
