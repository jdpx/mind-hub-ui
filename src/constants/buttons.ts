export const Primary = 'primary'
export const Secondary = 'secondary'
