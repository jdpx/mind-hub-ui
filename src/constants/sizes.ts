export const Small = 'sm'
export const Medium = 'md'
export const Large = 'lg'
export const ExtraLarge = 'xl'
