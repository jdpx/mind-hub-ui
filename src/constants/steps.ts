export const Video = 'Video'
export const Audio = 'Audio'
export const Question = 'Question'
