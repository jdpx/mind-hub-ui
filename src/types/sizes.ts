import { Small, Medium, Large, ExtraLarge } from '../constants/sizes'

export type Sizes = typeof Small | typeof Medium | typeof Large | typeof ExtraLarge
