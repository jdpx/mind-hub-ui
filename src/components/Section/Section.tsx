import React from 'react'
import classNames from 'classnames'

import { ExtraLarge, Large, Medium, Small } from '../../constants/sizes'
import './Section.scss'
import { Sizes } from '../../types/sizes'

interface Props {
    children: React.ReactNode
    size?: Sizes
    testid?: string
    className?: string
}

export function Section({ children, size = Small, testid, className }: Props) {
    const wrapperClass = classNames(
        'content-wrapper',
        {
            sm: size === Small,
            md: size === Medium,
            lg: size === Large,
            xl: size === ExtraLarge,
        },
        className,
    )

    return (
        <div className={wrapperClass} data-testid={testid}>
            {children}
        </div>
    )
}
export default Section
