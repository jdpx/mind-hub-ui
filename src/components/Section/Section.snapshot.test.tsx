import React from 'react'
import { render } from '@testing-library/react'
import Section from './Section'
import { ExtraLarge, Large, Medium, Small } from '../../constants/sizes'

describe('Section Snapshots', () => {
    it('renders section with child', () => {
        const { asFragment } = render(
            <Section>
                <div>Testing</div>
            </Section>,
        )

        expect(asFragment()).toMatchSnapshot()
    })

    it('renders section with title', () => {
        const { asFragment } = render(
            <Section title="Testing title">
                <div>Testing</div>
            </Section>,
        )

        expect(asFragment()).toMatchSnapshot()
    })

    it('renders small section', () => {
        const { asFragment } = render(
            <Section size={Small}>
                <div>Testing</div>
            </Section>,
        )

        expect(asFragment()).toMatchSnapshot()
    })

    it('renders Medium section', () => {
        const { asFragment } = render(
            <Section size={Medium}>
                <div>Testing</div>
            </Section>,
        )

        expect(asFragment()).toMatchSnapshot()
    })

    it('renders Large section', () => {
        const { asFragment } = render(
            <Section size={Large}>
                <div>Testing</div>
            </Section>,
        )

        expect(asFragment()).toMatchSnapshot()
    })

    it('renders ExtraLarge section', () => {
        const { asFragment } = render(
            <Section size={ExtraLarge}>
                <div>Testing</div>
            </Section>,
        )

        expect(asFragment()).toMatchSnapshot()
    })

    it('renders disabled section', () => {
        const { asFragment } = render(
            <Section disabled>
                <div>Testing</div>
            </Section>,
        )

        expect(asFragment()).toMatchSnapshot()
    })

    it('renders section with custom class', () => {
        const { asFragment } = render(
            <Section className="customClass">
                <div>Testing</div>
            </Section>,
        )

        expect(asFragment()).toMatchSnapshot()
    })

    it('renders section with testid', () => {
        const { asFragment } = render(
            <Section testid="test-id">
                <div>Testing</div>
            </Section>,
        )

        expect(asFragment()).toMatchSnapshot()
    })
})
