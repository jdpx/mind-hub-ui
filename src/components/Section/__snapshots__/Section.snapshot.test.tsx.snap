// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section Snapshots renders ExtraLarge section 1`] = `
<DocumentFragment>
  <div
    class="content-wrapper xl"
  >
    <div>
      Testing
    </div>
  </div>
</DocumentFragment>
`;

exports[`Section Snapshots renders Large section 1`] = `
<DocumentFragment>
  <div
    class="content-wrapper lg"
  >
    <div>
      Testing
    </div>
  </div>
</DocumentFragment>
`;

exports[`Section Snapshots renders Medium section 1`] = `
<DocumentFragment>
  <div
    class="content-wrapper md"
  >
    <div>
      Testing
    </div>
  </div>
</DocumentFragment>
`;

exports[`Section Snapshots renders disabled section 1`] = `
<DocumentFragment>
  <div
    class="content-wrapper disabled"
  >
    <div>
      Testing
    </div>
  </div>
</DocumentFragment>
`;

exports[`Section Snapshots renders section with child 1`] = `
<DocumentFragment>
  <div
    class="content-wrapper"
  >
    <div>
      Testing
    </div>
  </div>
</DocumentFragment>
`;

exports[`Section Snapshots renders section with custom class 1`] = `
<DocumentFragment>
  <div
    class="content-wrapper customClass"
  >
    <div>
      Testing
    </div>
  </div>
</DocumentFragment>
`;

exports[`Section Snapshots renders section with testid 1`] = `
<DocumentFragment>
  <div
    class="content-wrapper"
    data-testid="test-id"
  >
    <div>
      Testing
    </div>
  </div>
</DocumentFragment>
`;

exports[`Section Snapshots renders section with title 1`] = `
<DocumentFragment>
  <div
    class="content-wrapper withTitle"
  >
    <h3
      class="section-title"
    >
      Testing title
    </h3>
    <div>
      Testing
    </div>
  </div>
</DocumentFragment>
`;

exports[`Section Snapshots renders small section 1`] = `
<DocumentFragment>
  <div
    class="content-wrapper sm"
  >
    <div>
      Testing
    </div>
  </div>
</DocumentFragment>
`;
